app/grandchallenge/evaluation/tasks.py

```diff
@@ -49,8 +49,15 @@ def prepare_and_execute_evaluation(
     evaluation = Evaluation.objects.get(pk=evaluation_pk)
     submission = evaluation.submission
 
+    if not evaluation.additional_inputs_complete:
+        logger.info("Nothing to do, inputs are still being validated.")
+        return
+
     if evaluation.status != evaluation.VALIDATING_INPUTS:
-        raise RuntimeError("Evaluation is not ready to be executed.")
+        # this task can be called multiple times with complete inputs,
+        # and might have been queued for execution already, so ignore
+        logger.info("Evaluation has already been scheduled for execution.")
+        return
```

Comment:
Are you certain that this is the correct behaviour and not only works for the happy path? Previously we raised the exception here as it indicated that something went wrong with the celery task ordering.